pallet/src/lib.rs

```diff
@@ -156,11 +168,15 @@ pub mod pallet {
 
     impl<T: Config> SessionManager<T::AccountId> for AlephSessionManager<T> {
         fn new_session(session: u32) -> Option<Vec<T::AccountId>> {
-            if let Some((validators_list, validators_list_session)) = Pallet::<T>::validators_list()
+            if let Some(session_for_validators_change) =
+                Pallet::<T>::session_for_validators_change()
             {
-                if validators_list_session <= session {
-                    ValidatorsList::<T>::put(None::<(Vec<T::AccountId>, u32)>);
-                    return Some(validators_list);
+                if session_for_validators_change <= session {
+                    if let Some(validators) = Pallet::<T>::validators() {
```

Comment:
There should be either a pair of `Some`s or a pair of `None`'s (or at least that's what I would think), so it's better to use `expect` in place of `if let`.

Reply:
Done